packages/jaeger-ui/src/model/link-patterns.js

```diff
@@ -0,0 +1,215 @@
+// @flow
+
+// Copyright (c) 2017 The Jaeger Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import _uniq from 'lodash/uniq';
+import { getConfigValue } from '../utils/config/get-config';
+import type { Span, Trace } from '../types';
+
+const parameterRegExp = /\$\{([^{}]*)\}/g;
+
+type ProcessedTemplate = {
+  parameters: string[],
+  template: (data: { [parameter: string]: any }) => string,
+};
+
+function getParamNames(str) {
+  const names = [];
+  str.replace(parameterRegExp, (match, name) => {
+    names.push(name);
+    return match;
+  });
+  return _uniq(names);
+}
+
+function stringSupplant(str, encodeFn: any => string, map) {
+  return str.replace(parameterRegExp, (_, name) => encodeFn(map[name]));
```

Comment:
`encodeUriComponent` will stringify null (and undefined?) values, so maybe it makes sense to guard the encode function invocation?

```js
encodeURIComponent(null)
// -> "null"

fn = () => null;
'abbc'.replace(/b/g, fn)
// -> "anullnullc"
```

Reply:
Ok. As links are not displayed if any parameter is missing, this problem only happens when tags are explicitly added with a null or undefined value. In this case, it is true that is better to guard the encode function invocation and have an empty string rather than the strange "undefined" or "null" strings. I have updated the code.